src/strategy.ts

```diff
@@ -12,23 +12,39 @@ export default class SnykOAuth2Strategy extends Strategy {
    * profileFunc: is called is user wants to call any Snyk API
    * to get profile for their records
    */
-  private _nonce: string;
+  private _nonce = '';
+  private _tenantId = '';
   private _profileFunc?: ProfileFunc;
 
   constructor(options: SnykStrategyOptions, verify: VerifyFunctionWithRequest) {
     super(options, verify);
-    this._nonce = options.nonce;
```

Comment:
Shouldn't we keep supporting also the nonce from the options?

Reply:
Setting it through the constructor means it is the same for all requests going out, which defeats the purpose of `nonce`.